tapchannel/aux_traffic_shaper.go

```diff
@@ -0,0 +1,242 @@
+package tapchannel
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightninglabs/taproot-assets/address"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/rfq"
+	"github.com/lightninglabs/taproot-assets/rfqmsg"
+	cmsg "github.com/lightninglabs/taproot-assets/tapchannelmsg"
+	lfn "github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+var (
+	// DefaultOnChainHtlcAmount is the default amount that we consider
+	// as the smallest HTLC amount that can be sent on-chain. This needs to
+	// be greater than the dust limit for an HTLC.
+	DefaultOnChainHtlcAmount = lnwallet.DustLimitForSize(
+		input.UnknownWitnessSize,
+	)
+)
+
+// TrafficShaperConfig defines the configuration for the auxiliary traffic
+// shaper.
+type TrafficShaperConfig struct {
+	ChainParams *address.ChainParams
+
+	RfqManager *rfq.Manager
+}
+
+// AuxTrafficShaper is a Taproot Asset auxiliary traffic shaper that can be used
+// to make routing decisions for Taproot Asset channels.
+type AuxTrafficShaper struct {
+	startOnce sync.Once
+	stopOnce  sync.Once
+
+	cfg *TrafficShaperConfig
+
+	// ContextGuard provides a wait group and main quit channel that can be
+	// used to create guarded contexts.
+	*fn.ContextGuard
+}
+
+// NewAuxTrafficShaper creates a new Taproot Asset auxiliary traffic shaper
+// based on the passed config.
+func NewAuxTrafficShaper(cfg *TrafficShaperConfig) *AuxTrafficShaper {
+	return &AuxTrafficShaper{
+		cfg: cfg,
+		ContextGuard: &fn.ContextGuard{
+			DefaultTimeout: DefaultTimeout,
+			Quit:           make(chan struct{}),
+		},
+	}
+}
+
+// Start attempts to start a new aux traffic shaper.
+func (s *AuxTrafficShaper) Start() error {
+	var startErr error
+	s.startOnce.Do(func() {
+		log.Info("Starting aux traffic shaper")
+	})
+	return startErr
+}
+
+// Stop signals for an aux traffic shaper to gracefully exit.
+func (s *AuxTrafficShaper) Stop() error {
+	var stopErr error
+	s.stopOnce.Do(func() {
+		log.Info("Stopping aux traffic shaper")
+
+		close(s.Quit)
+		s.Wg.Wait()
+	})
+
+	return stopErr
+}
+
+// A compile-time check to ensure that AuxTrafficShaper fully implements the
+// routing.TlvTrafficShaper interface.
+var _ routing.TlvTrafficShaper = (*AuxTrafficShaper)(nil)
+
+// HandleTraffic is called in order to check if the channel identified by the
+// provided channel ID may have external mechanisms that would allow it to carry
+// out the payment.
+func (s *AuxTrafficShaper) HandleTraffic(_ lnwire.ShortChannelID,
+	fundingBlob lfn.Option[tlv.Blob]) (bool, error) {
```

Comment:
godoc vs function signature are kind of contradicting. We seem to not actually care about the short channel id?

Reply:
Yeah, we weren't really sure what information we actually need for each of the interfaces. They evolved over time. We can clean this up at a later point, because now the definition is already merged in `lnd`.